### PR TITLE
chore(dot/core, dot/sync, lib/babe): refactor handling on block import to reduce duplicate code

### PR DESCRIPTION
**Pull request by noot**
## Changes

- when a block is imported into the chain, either via from syncing a block from the network or importing a block we produced, various updates and checks need to happen, namely:
1. storing block in db
2. storing state trie in db
3. checking for runtime changes
4. checking for code substitutions
5. handling consensus digests
- previously this was happening in two different places (sync and babe) and in the case of babe some checks weren't happening at all
- this PR refactors the handling of an imported block to go through `core.Service` via the `HandleBlockImport` or `HandleBlockProduced` funtions for `sync` and `babe` respectively
- also, moved `core.DigestHandler` into its own package `dot/digest`:  `core.DigestHandler`  ->  `digest.Handler`
- cleanup relics of interfaces no longer needed by core, sync, and babe

## Tests

```
go test ./dot/core
go test ./dot/sync -short
go test ./lib/babe
```

## Issues

- related to #1595, related to #1624

### CODE REVIEW

**Review by EclesioMeloJunior — APPROVED** [review]

File: dot/rpc/modules/author_test.go [hunk 1]
```diff
@@ -132,10 +132,13 @@ func TestAuthorModule_SubmitExtrinsic_invalid_input(t *testing.T) {
 	txQueue := state.NewTransactionState()
 	auth := setupAuthModule(t, txQueue)
 
+	fmt.Println("setupAuthModule")
+
 	// create and submit extrinsic
 	ext := Extrinsic{fmt.Sprintf("%x", "1")}
 
 	res := new(ExtrinsicHashResponse)
+	fmt.Println("SubmitExtrinsic")
```
Comment (edwardmack) [inline dot/rpc/modules/author_test.go:141]: remove Println?

File: dot/rpc/modules/author_test.go [hunk 2]
```diff
@@ -276,21 +279,23 @@ func newCoreService(t *testing.T, srvc *state.Service) *core.Service {
...
+		CodeSubstitutedState: srvc.Base,
 	}
 
 	return core.NewTestService(t, cfg)
 }
 
 func setupAuthModule(t *testing.T, txq *state.TransactionState) *AuthorModule {
+	fmt.Println("calling setupAuthModule")
 	cs := newCoreService(t, nil)
+	fmt.Println("called newCoreService")
```
Comment (edwardmack) [inline dot/rpc/modules/author_test.go:298]: Remove println.

File: dot/sync/syncer.go [hunk 3]
```diff
@@ -27,24 +27,23 @@ import (
...
+
 	log "github.com/ChainSafe/log15"
 )
 
 var logger = log.New("pkg", "sync")
 
 // Service deals with chain syncing by sending block request messages and watching for responses.
 type Service struct {
-	codeHash common.Hash // cached hash of runtime code
+	//codeHash common.Hash // cached hash of runtime code
```
Comment (edwardmack) [inline dot/sync/syncer.go:39]: remove.

**Review by edwardmack — APPROVED** [review]
Nice work, lgtm.

**Pull request merged**